app/src/main/java/io/pslab/widgets/loggedData/logDataWidgetList.java

```diff
@@ -0,0 +1,155 @@
+package io.pslab.widgets.loggedData;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.v7.widget.DividerItemDecoration;
+import android.support.v7.widget.LinearLayoutManager;
+import android.view.View;
+import android.widget.RemoteViews;
+import android.widget.RemoteViewsService;
+
+import io.pslab.R;
+import io.pslab.adapters.SensorLoggerListAdapter;
+import io.pslab.models.PSLabSensor;
+import io.pslab.models.SensorDataBlock;
+import io.pslab.others.CSVLogger;
+import io.pslab.others.LocalDataLog;
+import io.realm.Realm;
+import io.realm.RealmResults;
+
+public class logDataWidgetList implements RemoteViewsService.RemoteViewsFactory {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Class names should start with an upper case letter (NM_CLASS_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-should-start-with-an-upper-case-letter-nm-class-naming-convention).
Class names should be nouns, in mixed case with the first letter of each internal word capitalized. Try to keep your class names simple and descriptive. Use whole words-avoid acronyms and abbreviations (unless the abbreviation is much more widely used than the long form, such as URL or HTML).